common/helpers/src/avro.rs

```diff
@@ -0,0 +1,248 @@
+/// # Returns
+/// * `Result<(Vec<Schema>, Vec<Vec<u8>>), AvroError>` - structured and serialized Avro schemas
+/// # Examples
+/// ```
+/// use avro::*;
+/// let raw_schema = r#"{"type": "record", "name": "User", "fields": [{"name": "name", "type": "string"}, {"name": "favorite_number", "type": "int"}]}"#;
+/// let schema_result = fingerprint_raw_schema_list(vec![raw_schema.to_string()]);
+/// assert!(schema_result.is_ok());
+/// let serialized_schemas = schema_result.unwrap().1;
+/// ```
+pub fn fingerprint_raw_schema_list(
+	raw_schema: &[&str],
+) -> Result<(Vec<Schema>, Vec<Vec<u8>>), AvroError> {
+	let mut schema_fingerprints = Vec::new();
+	let mut schemas = Vec::new();
+	for s in raw_schema {
+		let (schema, fingerprint) = fingerprint_raw_schema(s)?;
+		schema_fingerprints.push(fingerprint);
+		schemas.push(schema);
+	}
+	Ok((schemas, schema_fingerprints))
+}
+
+///Function to convert a serialized Avro schema into Avro Schema type.
+/// If schema is malformed or invalid, returns an error.
+/// # Arguments
+/// * `serialized_schema` - serialized Avro schema to be converted
+/// # Returns
+/// * `Result<Schema, AvroError>` - structured Avro schema
+/// # Examples
+/// ```
+/// use avro::*;
+/// let raw_schema = r#"{"type": "record", "name": "User", "fields": [{"name": "name", "type": "string"}, {"name": "favorite_number", "type": "int"}]}"#;
+/// let serialized_schema = fingerprint_raw_schema(raw_schema);
+/// assert!(serialized_schema.is_ok());
+/// let schema = schema_result.unwrap().1;
+/// let translated_schema = translate_schema(&schema);
+/// assert!(translated_schema.is_ok());
+/// ```
+pub fn translate_schema(serialized_schema: Vec<u8>) -> Result<Schema, AvroError> {
+	let schema_str = str::from_utf8(&serialized_schema);
+	match schema_str {
+		Ok(schema_str) => {
+			let schema = Schema::parse_str(schema_str)?;
+			Ok(schema)
+		},
+		Err(error) => Err(AvroError::InvalidSchema(error.to_string())),
+	}
+}
+
+///Function to convert a list of serialized Avro schema into Avro Schema type.
+/// If schema is malformed or invalid, returns an error.
+/// # Arguments
+/// * `serialized_schemas` - list of serialized Avro schema to be converted
+/// # Returns
+/// * `Result<Vec<Schema>, AvroError>` - structured Avro schema
+/// # Examples
+/// ```
+/// use avro::*;
+/// let raw_schema = r#"{"type": "record", "name": "User", "fields": [{"name": "name", "type": "string"}, {"name": "favorite_number", "type": "int"}]}"#;
+/// let serialized_schema = fingerprint_raw_schema(raw_schema);
+/// assert!(serialized_schema.is_ok());
+/// let schema = schema_result.unwrap().1;
+/// let translated_schema = translate_schemas(&vec![schema]);
+/// assert!(translated_schema.is_ok());
+/// ```
+pub fn translate_schemas(serialized_schemas: Vec<Vec<u8>>) -> Result<Vec<Schema>, AvroError> {
+	let mut schemas = Vec::new();
+	for s in serialized_schemas {
+		schemas.push(translate_schema(s)?);
+	}
+	Ok(schemas)
+}
+
```

Comment:
same as previous this time translate a list of serialized avro schema to Avro Schema type

Reply:
using rayon parallel iterators